### PR TITLE
Add reverse merge

### PR DESCRIPTION
**Pull request by yiiliveext**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️

Main app events config
```php
use Yiisoft\Arrays\Modifier\MergeReverse;
use Yiisoft\Yii\Web\Event\ApplicationStartup;
use App\StartupCollector;

return [
    ApplicationStartup::class => [
        [StartupCollector::class, 'collect'],
    ],
    MergeReverse::class => new MergeReverse(),
];
```
Package events config
```php
use Yiisoft\Arrays\Modifier\MergeReverse;
use Yiisoft\Yii\Web\Event\ApplicationStartup;
use Yiisoft\Yii\Web\Event\ApplicationShutdown;

return [
    ApplicationStartup::class => [
        [WebAppInfoCollector::class, 'collect'],
    ],
   ApplicationShutdown::class => [
        [WebAppInfoCollector::class, 'collect'],
    ],
];
```
Result
```php
return [
    'Yiisoft\\Yii\\Web\\Event\\ApplicationStartup' => [
        ['App\\StartupCollector', 'collect'],
        ['Yiisoft\\Yii\\Debug\\Collector\\WebAppInfoCollector', 'collect'],
    ],
    'Yiisoft\\Yii\\Web\\Event\\ApplicationShutdown' => [
        ['Yiisoft\\Yii\\Debug\\Collector\\WebAppInfoCollector', 'collect']
    ],

```
```

### CODE REVIEW

File: src/Modifier/MergeReverse.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Yiisoft\Arrays\Modifier;
+
+/**
+ * Reverse arrays order
+ */
+class MergeReverse implements ModifierInterface
```
Comment (samdark) [inline src/Modifier/MergeReverse.php:8]: ReverseBlockMerge

**Comment by samdark** [comment]
Need to remove ReverseValues. It was meant for the same use-case but didn't work well.

**Comment by samdark** [comment]
Good one 👍

**Comment by hiqsol** [comment]
Why remove `ArrayReverse`? We're using it and it does it's work good!

**Pull request merged**